internal/controllers/provisioningrequest_controller.go

```diff
@@ -161,16 +162,21 @@ func (t *provisioningRequestReconcilerTask) run(ctx context.Context) (ctrl.Resul
 		return res, err
 	}
 
+	unstructuredClusterInstance, err := utils.ConvertToUnstructured(*renderedClusterInstance)
```

Comment:
Perhaps add a TODO here to indicate that the handlePreProvisioning function should be updated to return an unstructured ClusterInstance.